### DISCUSSION TITLE
If the callback that defines the store is an async function, the return value obtained externally will be undefined. Why?

### DISCUSSION THREAD

**Discussion by ugCrying** [vuejs/pinia#2294] [Help and Questions] [answered]
<img width="539" alt="image" src="https://github.com/vuejs/pinia/assets/53458469/e198eaf6-7e4a-42b9-90e2-818c8fbdb391">

<img width="498" alt="image" src="https://github.com/vuejs/pinia/assets/53458469/26021026-1212-4610-a105-9efdc1ceb4d1">

<img width="367" alt="image" src="https://github.com/vuejs/pinia/assets/53458469/e2804f84-0fe6-426a-b574-34506fc6cf42">

Does it not support async function? No corresponding description was found in the documentation.

**Comment by posva** [comment, marked as answer]
The function of a setup store cannot be async, it must return an object with the properties (https://pinia.vuejs.org/core-concepts/#Setup-Stores) not a promise

**Reply by n0099** [reply]
It would be nice if there's an type error when doing so.
